### ISSUE TITLE
Floating point exception (core dumped) when running inference demo

### ISSUE THREAD

**Issue by ArixLi** [microsoft/Magma#41]
First thanks for your great work!

When I am trying to run the inference demo (also every inference demo like agent or robot) you provide in the README with local model files, I encounter 

> Floating point exception (core dumped)

My env is:
torch==2.3.1+CUDA12.1 on H20 GPUs, transformers==4.44.1 with specific version you provide, and flash_attn                   2.7.4.post1

`
[Magma]# python inference.py 
2025-03-11 16:07:32.705447: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-03-11 16:07:32.719562: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1741680452.735489  426285 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1741680452.740242  426285 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-03-11 16:07:32.757915: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI AVX512_BF16 AVX512_FP16 AVX_VNNI AMX_TILE AMX_INT8 AMX_BF16 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
INFO:magma.image_tower_magma:Loaded hf-hub:laion/CLIP-convnext_xxlarge-laion2B-s34B-b82K-augreg model config.
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:01<00:00,  3.42it/s]
/root/miniconda3/envs/sr-simplerenv/lib/python3.10/site-packages/transformers/models/auto/image_processing_auto.py:513: FutureWarning: The image_processor_class argument is deprecated and will be removed in v4.42. Please use `slow_image_processor_class`, or `fast_image_processor_class` instead
  warnings.warn(
/root/miniconda3/envs/sr-simplerenv/lib/python3.10/site-packages/torch/utils/checkpoint.py:464: UserWarning: torch.utils.checkpoint: the use_reentrant parameter should be passed explicitly. In version 2.4 we will raise an exception if use_reentrant is not passed. use_reentrant=False is recommended, but if you need to preserve the current default behavior, you can pass use_reentrant=True. Refer to docs for more details on the differences between the two variants.
  warnings.warn(
/root/miniconda3/envs/sr-simplerenv/lib/python3.10/site-packages/torch/utils/checkpoint.py:91: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
We detected that you are passing `past_key_values` as a tuple and this is deprecated and will be removed in v4.43. Please use an appropriate `Cache` class (https://huggingface.co/docs/transformers/v4.41.3/en/internal/generation_utils#transformers.Cache)
Floating point exception (core dumped)`

**Comment by ArixLi** [comment]
This can be solved by setting `dtype=torch.float32` or pip install nvidia-cublas-cu12==12.4.5.8 for torch==2.3.1+cuda121

**Comment by jwyang** [comment]
I think this is because your GPU does not support bfloat16, maybe you have to use float16 or float32 as you tried above.
